Voycar.Api.Web.Tests.Integration/Users/Post_Register/Tests.cs

```diff
@@ -0,0 +1,327 @@
+namespace Voycar.Api.Web.Tests.Integration.Users.Post_Register;
+
+using Context;
+using Entities;
+using Microsoft.EntityFrameworkCore;
+using R = Features.Users.Endpoints.Post.Register;
+
+
+public sealed class State : StateFixture
+{
+    public Guid Id { get; set; }
+
+    public static Guid RoleId { get; set; }
+    public const string RoleName = "member";
+
+    public static Guid PlanId { get; set; }
+    public const string PlanName = "basic";
+
+    public R.Request Request { get; set; } = CreateValidRequest();
+
+
+    public static R.Request CreateValidRequest()
+    {
+        return new R.Request
+        {
+            Email = "test@test.de",
+            Password = "notsafe987",
+            FirstName = "null",
+            LastName = "null",
+            Street = "null",
+            HouseNumber = "null",
+            PostalCode = "null",
+            City = "null",
+            Country = "null",
+            BirthDate = new DateOnly(2002, 12, 12),
+            BirthPlace = "null",
+            PhoneNumber = "null",
+            PlanId = PlanId
+        };
+    }
+}
+
+
+public class Tests : TestBase<App, State>
+{
+    private readonly App _app;
+    private readonly State _state;
+    private readonly VoycarDbContext Context;
+
+
+    // Setup request client
+    public Tests(App app, State state)
+    {
+        this._app = app;
+        this.Context = this._app.Context;
+        this._state = state;
+        State.RoleId = this.Context.Roles.First(r => r.Name == State.RoleName).Id;
+        State.PlanId = this.Context.Plans.First(p => p.Name == State.PlanName).Id;
+    }
+
+
+    [Fact]
+    public async Task Post_Request_ReturnsOk_And_SavesUserInDb()
+    {
+        // Arrange
+        var request = State.CreateValidRequest();
+
+        // Act
+        var httpResponse = await this._app.Client.POSTAsync<R.Endpoint, R.Request>(request);
+
+        // Arrange assertion
+        var userInDb = await this.Context.Users.FirstOrDefaultAsync(user => user.Email == request.Email);
+        var memberInDb = await this.Context.Members.FirstOrDefaultAsync(member => member.Id == userInDb!.MemberId);
+
+        // Assert
+        httpResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+        userInDb.Should().NotBeNull();
+        memberInDb.Should().NotBeNull();
```

Comment:
Das `!.` nur nutzen, wenn wir auch garantieren können, dass der Value nicht `null` ist. In der einen Zeile gibt es dann eine Warnung, aber das ist richtig und ok. Wenn der Wert null ist, dann failed der Test halt.
```suggestion
        var memberInDb = await this.Context.Members.FirstOrDefaultAsync(member => member.Id == userInDb.MemberId);

        // Assert
        httpResponse.StatusCode.Should().Be(HttpStatusCode.OK);
        userInDb!.Should().NotBeNull();
        memberInDb!.Should().NotBeNull();
```

Reply:
Aber würde der Test nicht auch failen, wenn da ein `!` ist und der Wert `null` ist?

Reply:
Ich glaube schon. Da muss nicht unbedingt ein ! hin

Reply:
Ahh ups, ja habt ihr recht. Ich wollte die unteren beiden `!.` zwei Zeilen weiter unten machen 😅 (damit die Warnung weg ist)